crates/sui-core/src/consensus_adapter.rs

```diff
@@ -166,11 +167,19 @@ impl ConsensusAdapter {
 
         // Check if this authority submits the transaction to consensus.
         if Self::should_submit(certificate) {
-            self.consensus_client
+            let res = self
+                .consensus_client
                 .clone()
                 .submit_transaction(TransactionProto { transaction: bytes })
                 .await
-                .map_err(|e| SuiError::ConsensusConnectionBroken(format!("{:?}", e)))?;
+                .map_err(|e| SuiError::ConsensusConnectionBroken(format!("{:?}", e)));
+
+            // Record an internal error
+            if let Err(e) = &res {
+                error!("Submit transaction failed with: {:?}", e);
+            }
+
+            res?;
```

Comment:
The [`inspect`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6a0e7699c562331d986eba141089d519) method on result, or [tap](https://docs.rs/tap/1.0.1/tap/). is a nicer way to do this I think.